### DISCUSSION TITLE
ULA generation

### DISCUSSION THREAD

**Discussion by amira123bd** [NVlabs/sionna#366] [General]
i want to generate a ULA, is this correct using this configuration 
 
```py
scene.rx_array = PlanarArray(num_rows=64, num_cols=1,
                             vertical_spacing=0.5,
                             horizontal_spacing=0.5,
                             pattern="iso",
                             polarization="V")
```

**Comment by jhoydis** [comment]
Hi,

This would create a vertical ULA. If you want a horizontal ULA, you should use `num_cols=64` and `num_rows=1`.
Alternatively, you could apply a rotation via the `orientation` parameter of the receiver or transmitter..

**Reply by amira123bd** [reply]
With num_cols=64 and num_rows=1 create a horizontal ULA over the z axis , and with num_cols=1 and num_rows=64 create a vertical ULA over the y axis what if i want ULA that is aligned over the x-axis?

**Reply by jhoydis** [reply]
A planar array is always assumed to lie in the y-z plane with its normal pointing toward the positive x axis.
Please have a look at [this example](https://nvlabs.github.io/sionna/api/rt.html#sionna.rt.PlanarArray) from the documentation.
You can always use the class method [`Array.show()`](https://nvlabs.github.io/sionna/api/rt.html#sionna.rt.PlanarArray.show) to visualize an array.

Now, each `RadioDevice`, such as a `Transmitter` or `Receiver`, has an `orientation` property that will determine the orientation of its antenna array.

For example, if you want to have a `Receiver` that has an x-axis aligned horizontal ULA, you would do the following:

```
# Create a horizontal ULA that is aligned with the y-axis
scene.rx_array = PlanarArray(num_rows=1, num_cols=64,
                             vertical_spacing=0.5,
                             horizontal_spacing=0.5,
                             pattern="iso",
                             polarization="V")

# Create a receiver which is rotated by 90 degrees around the z-axis
# Its antenna array will point toward the positive y axis
scene.add(Receiver("rx", position=[0,0,0], orientation=[PI/2, 0, 0])
```
